src/compose.ts

```diff
@@ -13,6 +13,33 @@ export class ComposeError extends Error {
   }
 }
 
+class ComposeCommand {
+  private composeCommand = 'docker-compose';
+  private singletonCalled = false;
+
+  /**
+   * Checks docker-compose availability, using docker compose as a fallback.
+   *
+   * @returns {Promise<string>} 'docker-compose' or 'docker compose'
+   */
+  async get(): Promise<string> {
+    if (!this.singletonCalled) {
+      this.singletonCalled = true;
+      try {
+        await exec.exec('docker-compose', ['--version']);
+      } catch (error) {
+        console.warn(
+          'docker-compose not available, falling back to docker compose'
+        );
+        this.composeCommand = 'docker compose';
+      }
+    }
```

Comment:
Not using constructor for singleton because constructor cannot be async and there's the
```ts
await exec.exec('docker-compose', ['--version']);
```